crates/sui-network/src/state_sync/mod.rs

```diff
@@ -1148,11 +1225,17 @@ where
     S: WriteStore + Clone,
     <S as ReadStore>::Error: std::error::Error,
 {
-    let peers = PeerBalancer::new(&network, peer_heights.clone())
-        .with_checkpoint(*checkpoint.sequence_number());
+    let peers = PeerBalancer::new(
+        &network,
+        peer_heights.clone(),
+        PeerCheckpointRequestType::Content,
+    )
+    .with_checkpoint(*checkpoint.sequence_number());
     let Some(contents) = get_full_checkpoint_contents(peers, &store, &checkpoint, timeout).await else {
         // Delay completion in case of error so we don't hammer the network with retries.
-        tokio::time::sleep(Duration::from_secs(10)).await;
+        // TODO: we might want to connect to better peers here.
```

Comment:
I'd remove the TODO here because this part of the code would not be where you manage which peers to connect to - here it just iterates through all available peers whose connections were formed elsewhere